README.md

```diff
@@ -1,2 +1,578 @@
+                message: err.message
+            })
+        }
+    })
+```
+
+auto-block lets you declare these error mappings using `errorDefaults` and
+`errorMappings`:
+
+```javascript
+    var controller: {
+        block: {
+            'one': {
+                func: one
+            },
+            'two': {
+                func: two,
+                with: {
+                    'alpha': 'one.alpha',
+                    'beta': 'one.beta'
+                },
+                errorDefaults: {
+                    'status': 404
+                },
+                errorMappings: {
+                    'alpha': 'one.alpha'
+                }
+            }
+        },
+        done: function (err, results) {
+            if (err && err.status) {
+                res.status(err.status).send({
+                    message: err.message
+                })
+            }
+        }
+    }
+```
+
+These, along with other features, allow you to build complex controllers with
+needing to explicitly write function wrappers. auto-block does all of that for
+you.
+
+Without the need for extra wrappers, the temptation to inline business logic is
+removed. You can comfortably move the business logic out of the controller
+without needing to know the details of which controller module you used.
+
+## Controller configuration
+
+### .done
+
+```javascript
+    var controller = {
+        done: function (error, response) {
+            console.log('do stuff');
+        }
+    }
+```
+
+Called after the entire block has been completed. The values of `error` and
+`response` are generated by their respective mapping configurations (see
+below).
+
+The `done` function can also be provided as the second parameter to `autoBlock.run`:
+
+```javascript
+    autoBlock.run(controller, function (error, response) {
+        console.log('do stuff');
+    })
+```
+
+### .data
+
+```javascript
+    var controller = {
+        data: {
+            'foo': 'bar',
+            'fizz': 'buzz'
+        }
+    }
+```
+
+Values provided in the `data` fields are available during `options`, `error` and
+`response` mappings but _not_ during `results` mapping.
+
+### .block
+
+```javascript
+    controller = {
+        block: {
+            'alpha': {
+                // ...
+            },
+            'beta': {
+                // ...
+            }
+        }
+    }
+```
+
+`block` holds the actual steps used during `autoBlock.run`. See below for details on how to configure steps properly.
+
+### Option mapping
+
+The values in `.data` are not exposed to the individual steps. You can
+explicitly expose them, however, using `optionsMapping`:
+
+```javascript
+    controller = {
+        data: {
+            'foo': {
+                'bar': 'zaz'
+            }
+        },
+        optionsDefaults: {
+            'fizz': 'buzz'
+            'bar': 'not zaz'
+        },
+        optionsMapping: {
+            'bar': 'foo.bar'
+        },
```

Comment:
Yeah, it would be nice. I chose keeping it this way because I'd rather not have to detect object shapes.